src/client/Microsoft.Identity.Client/AppConfig/ManagedIdentityApplicationBuilder.cs

```diff
@@ -218,7 +216,7 @@ private void ComputeClientIdForCaching()
         {
             if (!string.IsNullOrEmpty(Config.ManagedIdentityUserAssignedClientId))
             {
-                Config.ClientId = Config.ManagedIdentityUserAssignedClientId;
+                Config.ClientId = Constants.ManagedIdentityDefaultClientId + Config.ManagedIdentityUserAssignedClientId;
```

Comment:
1. Consider having a shorter constant;
2. Consider not using a hash for resource id; let's wait for feedback on this.
3. Would like an ack from @rayluo on formatting the client_id in the cache schema, so that we're on the same page.